Lab6/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,97 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Animal implements Comparable<Animal> {
+
+
+    public int getId() {
+        return id;
+    }
+
+    private int id;
+    private Vector2d location;
+    private MapDirection orientation;
+    private IWorldMap map;
+    private List<IPositionChangeObserver> observerList;
+
+    // zrobic na odwrot konstruktory, to maly powinien "dziedziczyc" z zduzego
+    public Animal(IWorldMap map) {
+        this(map, new Vector2d(2,2));
+    }
+
+    public Animal(IWorldMap map, Vector2d initialPoistion) {
+        this.map = map;
+        this.location = initialPoistion;
+        this.orientation = MapDirection.NORTH;
+        this.observerList = new ArrayList<>();
+        this.addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.